### ISSUE TITLE
GraalVM not (yet) notarized for macOS Catalina

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#1724]
In the next release, GraalVM for macOS should be notarized. I'm not sure if published releases can be notarized in retrospect, but that'd be great too.

More information: https://developer.apple.com/news/?id=09032019a

Related: https://github.com/graalvm/homebrew-tap/issues/6

**Comment by fniephaus** [comment]
In the meantime, `xattr -d com.apple.quarantine $GRAALVM_HOME` should be enough to get you going on macOS Catalina, no need to disable Gatekeeper.

**Comment by machi1990** [comment]
Hi @fniephaus, I have an error `xattr: /xx/xx/graalvm-ce/Contents/Home: No such xattr: com.apple.quarantine` when running the command `xattr -d com.apple.quarantine $GRAALVM_HOME`.  Thoughts?

**Comment by fniephaus** [comment]
@machi1990 you might need to launch GraalVM once...and what if you run it against /xx/xx/graalvm-ce instead of the Home dir?

**Comment by machi1990** [comment]
> and what if you run it against /xx/xx/graalvm-ce instead of the Home dir?

With this I am not seeing the previous error but `“gu” cannot be opened because the developer cannot be verified`, like the command did not have any effect.

**Comment by fniephaus** [comment]
Looks like you actually have to run it recursively against GraalVM's root directory:
`xattr -r -d com.apple.quarantine /xx/xx/graalvm-ce-19.2.0.1`

**Comment by machi1990** [comment]
> Looks like you actually have to run it recursively against GraalVM's root directory:
> `xattr -r -d com.apple.quarantine /xx/xx/graalvm-ce-19.2.0.1`

Thanks, this seems to do the trick. Running the command as `sudo` might be required :-)

**Comment by fniephaus** [comment]
@machi1990 glad it's working for you now!

**Comment by machi1990** [comment]
Yeah, I am glad too. Let me update the PR that I just tagged you :-)

**Comment by alonsoir** [comment]
Hi, i think i have the same problem. I cannot run the command in Catalina 10.15.2:

bash-3.2$ ./gu install native-image
 
It returns me this message: cannot be opened because the developer cannot be verified.

I have tried running 
sudo xattr -r -d com.apple.quarantine $GRAALVM_HOME

where 

bash-3.2$ echo $GRAALVM_HOME
/Library/Java/JavaVirtualMachines/graalvm-ce-java8-19.3.0/Contents/

with no luck. I cannot run the command.

Help, please

**Comment by XanderAtBackboneSystems** [comment]
@alonsoir I dont known whether it solves you problem but as far as i known the `$GRAALVM_HOME` directory should point to the `Home` directory within `Contents`. Which is not the case in your echo:
```
bash-3.2$ echo $GRAALVM_HOME
/Library/Java/JavaVirtualMachines/graalvm-ce-java8-19.3.0/Contents/
```

And @fniephaus also [stated](https://github.com/oracle/graal/issues/1724#issuecomment-540160024) that the command should be executed on the root directory of graalvm: `/Library/Java/JavaVirtualMachines/graalvm-ce-java8-19.3.0`

**Comment by alonsoir** [comment]
Thank you @XanderAtBackboneSystems. I already know it, thats why i introduce the UPDATE message.

**Comment by radeusgd** [comment]
What is the status of this? Are there plans to notarize the Graal packages in the near future?

**Comment by mjaggard** [comment]
In 2019 @fniephaus wrote "In the next release, GraalVM for macOS should be notarized." but the community editions from homebrew at least are still not.

**Comment by rhanton** [comment]
Are they not in the same build pipeline? How do you notarize some and not others?
